File: frontend/src/types/categoria.ts
import type { FinalidadeCategoria } from "./enums";

export type CreateCategoriaRequest = {
    descricao: string;
    finalidade: FinalidadeCategoria;
};

export type CategoriaResponse = {
    categoriaId: number;
    descricao: string;
    finalidade: FinalidadeCategoria;
};

File: backend/TesteT.Api/Domain/Entities/Categoria.cs
using TesteT.Api.Domain.Enums;

namespace TesteT.Api.Domain.Entities
{
    public class Categoria
    {
        public int CategoriaId { get; set; }
        public string Descricao { get; set; } = string.Empty;
        public FinalidadeCategoria Finalidade { get; set; }
    }
}